cmd/shell/cmd/auth.go

```diff
@@ -0,0 +1,27 @@
+package cmd
+
+import "context"
+
+var Auth = &auth{}
+
+type auth struct {
+	Username string
+	Password string
+}
+
+// BasicAuthCredentials is a custom type implementing grpc.PerRPCCredentials.
+type basicAuthCredentials struct {
```

Comment:
Explain why we need this struct?

Reply:
Move this to `grpc/basicauth` package.
Make sure http module also use this package.